model/src/main/kotlin/VcsInfo.kt

```diff
@@ -124,8 +125,20 @@ data class VcsInfo(
 }
 
 class VcsInfoDeserializer : StdDeserializer<VcsInfo>(VcsInfo::class.java) {
+    companion object {
+        val KNOWN_FIELDS = listOf("type", "url", "revision", "resolvedRevision", "path", "data")
+    }
+
     override fun deserialize(p: JsonParser, ctxt: DeserializationContext): VcsInfo {
         val node = p.codec.readTree<JsonNode>(p)
+
+        val fields = node.fields().asSequence().map { it.key }.toList()
```

Comment:
Extracting this code to a separate function would keep
the deserialze function more readable, e.g. consider "checkNoUnknownFields()" , analog to checkNotNull style functions.

Reply:
I will extract this to a method in another PR when I implement the same check for the other deserializers.